Build failure: paper-plane

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```shell
nix-shell -p paper-plane
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
these 3 derivations will be built:
  /nix/store/n8ymzg8j60xr5jsax3szlwh56qv979wy-rlottie-paperplane-0-unstable-2022-09-14.drv
  /nix/store/z1v7b1v0k5lj529lvhv7vd2ajfxrsnb4-tdlib-paperplane-1.8.19.drv
  /nix/store/vp3w6zvlgy8gwkr4493h5lzpm58yf8b1-paper-plane-0.1.0-beta.5.drv
these 15 paths will be fetched (48.86 MiB download, 368.18 MiB unpacked):
  /nix/store/9ig2lm1h3pxcsbahs03vvibpx78bziy2-blueprint-compiler-0.18.0
  /nix/store/7cviqkb3v2213y7yhj3lhyva5ak6pfqx-gperf-3.3
  /nix/store/jzf8bqy7xfq8b520v2aimbsp2n6ijspj-gst-libav-1.26.5
  /nix/store/0iglsdiwygkw7285dqsc9kr11vrc543g-gst-libav-1.26.5-dev
  /nix/store/5sjzsvgqn6hkhs4i3an68675kvfbfl9h-gtk4-4.18.6
  /nix/store/lia4wk9s5m8xrxsp1h6chr22mnkscnv2-gtk4-4.18.6-dev
  /nix/store/qhxld3z9crdqa2snqc4j6vdivx203www-libadwaita-1.7.6
  /nix/store/jdkar7grk454pflvjylxj56mgqrdczfx-libadwaita-1.7.6-dev
  /nix/store/w9w9wc5l3nwriribnqrgyx7mizm1fvk7-libshumate-1.4.0
  /nix/store/79z1ibynvrr8p13qbkp1zp3hjz3fii61-libshumate-1.4.0-dev
  /nix/store/2gw2cqlda39pmy4aafqpxz6s8p0kzsf9-paper-plane-0.1.0-beta.5-vendor
  /nix/store/6b37q22i0mzjllyqwsmr78rrv08b21ya-source
  /nix/store/jwy2dafjcxd73rd9jvnkahkcdrzx1b0g-source
  /nix/store/pnsb7gmvwhbynn7hncc5nlmxclmzbhma-source
  /nix/store/w9c2a6p8ka7lhazs5q1jcnpsbx02l1vw-wrap-gapps-hook
copying path '/nix/store/2gw2cqlda39pmy4aafqpxz6s8p0kzsf9-paper-plane-0.1.0-beta.5-vendor' from 'https://cache.nixos.org'...
copying path '/nix/store/pnsb7gmvwhbynn7hncc5nlmxclmzbhma-source' from 'https://cache.nixos.org'...
copying path '/nix/store/9ig2lm1h3pxcsbahs03vvibpx78bziy2-blueprint-compiler-0.18.0' from 'https://cache.nixos.org'...
copying path '/nix/store/jzf8bqy7xfq8b520v2aimbsp2n6ijspj-gst-libav-1.26.5' from 'https://cache.nixos.org'...
copying path '/nix/store/w9w9wc5l3nwriribnqrgyx7mizm1fvk7-libshumate-1.4.0' from 'https://cache.nixos.org'...
copying path '/nix/store/5sjzsvgqn6hkhs4i3an68675kvfbfl9h-gtk4-4.18.6' from 'https://cache.nixos.org'...
copying path '/nix/store/6b37q22i0mzjllyqwsmr78rrv08b21ya-source' from 'https://cache.nixos.org'...
copying path '/nix/store/7cviqkb3v2213y7yhj3lhyva5ak6pfqx-gperf-3.3' from 'https://cache.nixos.org'...
copying path '/nix/store/jwy2dafjcxd73rd9jvnkahkcdrzx1b0g-source' from 'https://cache.nixos.org'...
copying path '/nix/store/0iglsdiwygkw7285dqsc9kr11vrc543g-gst-libav-1.26.5-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/79z1ibynvrr8p13qbkp1zp3hjz3fii61-libshumate-1.4.0-dev' from 'https://cache.nixos.org'...
building '/nix/store/z1v7b1v0k5lj529lvhv7vd2ajfxrsnb4-tdlib-paperplane-1.8.19.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/jwy2dafjcxd73rd9jvnkahkcdrzx1b0g-source
source root is source
Running phase: patchPhase
substituteStream() in derivation tdlib-paperplane-1.8.19: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'CMake/GeneratePkgConfig.cmake')
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/doc/TDLib -DCMAKE_INSTALL_INFODIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/include -DCMAKE_INSTALL_SBINDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19 -DTD_E2E_ONLY:BOOL=FALSE
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
error: Cannot build '/nix/store/z1v7b1v0k5lj529lvhv7vd2ajfxrsnb4-tdlib-paperplane-1.8.19.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19
       Last 19 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/jwy2dafjcxd73rd9jvnkahkcdrzx1b0g-source
       > source root is source
       > Running phase: patchPhase
       > substituteStream() in derivation tdlib-paperplane-1.8.19: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'CMake/GeneratePkgConfig.cmake')
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/doc/TDLib -DCMAKE_INSTALL_INFODIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/include -DCMAKE_INSTALL_SBINDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/iznnldybnglhi68ldgxbzqcsvdpmc61m-tdlib-paperplane-1.8.19 -DTD_E2E_ONLY:BOOL=FALSE
       > CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       > 
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix log /nix/store/z1v7b1v0k5lj529lvhv7vd2ajfxrsnb4-tdlib-paperplane-1.8.19.drv
error: Cannot build '/nix/store/vp3w6zvlgy8gwkr4493h5lzpm58yf8b1-paper-plane-0.1.0-beta.5.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/16l73lf5ljpr8w32ayln48hwl7fn5m8s-paper-plane-0.1.0-beta.5
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.12, NixOS, 25.11 (Xantusia), 25.11.20251020.bc67cbb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/fnhg8x3zv2xsc4h8mkxg56aa66kh8wps-source`

### Notify maintainers

@Aleksanaa

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pa/paper-plane/package.nix
+++ b/pkgs/by-name/pa/paper-plane/package.nix
@@ -53,6 +53,12 @@ let
       rev = "2589c3fd46925f5d57e4ec79233cd1bd0f5d0c09";
       hash = "sha256-mbhxuJjrV3nC8Ja7N0WWF9ByHovJLmoLLuuzoU4khjU=";
     };
+    postPatch = ''
+      substituteInPlace CMakeLists.txt \
+        --replace-fail "cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)" "cmake_minimum_required(VERSION 3.10)"
+      substituteInPlace td/generate/tl-parser/CMakeLists.txt \
+        --replace-fail "cmake_minimum_required(VERSION 3.0 FATAL_ERROR)" "cmake_minimum_required(VERSION 3.10)"
+    '';
   });
   rlottie-paperplane = rlottie.overrideAttrs (prev: {
     pname = "rlottie-paperplane";

